src/pkg/storage/storage.go

```diff
@@ -36,6 +36,11 @@ const (
 type Storage struct {
 	Provider storageProvider
 	Config   map[string]string
+	// TODO: These are AWS S3 specific -> move these out
```

Comment:
per this todo: reasons not to move them to `/storage/s3.go` right away?  Extending the `S3Config` struct (or adding a S3ConfigWithRole struct) would also also save the effort of wiring up the `NewStorageUsingRole` func below, since NewStorage would do everything you'd need.